src/python/pants/pantsd/service/pailgun_service.py

```diff
@@ -80,11 +81,17 @@ def runner_factory(sock, arguments, environment):
         arguments,
         environment,
         graph_helper,
-        self.lock,
+        self.fork_lock,
         deferred_exc
       )
 
-    return PailgunServer(self._bind_addr, runner_factory)
+    # Plumb the daemon's lifecycle lock to the `PailgunServer` to safeguard teardown.
+    @contextmanager
+    def context_lock():
```

Comment:
Ditto consistent naming... helps with readability in the absence of types to distinguish these locks from one another.